### ISSUE TITLE
hub submodule is not working via terraform

### ISSUE THREAD

**Issue by guid1984** [terraform-google-modules/terraform-google-kubernetes-engine#1132]
### TL;DR

the hub submodule is not working, specifically the script gke_hub_registration.sh. Getting permission denied error when terraform trying to execute the sh script via "terraform-google-modules/gcloud/google" 

![image](https://user-images.githubusercontent.com/8578605/150882855-e66a9b21-baa1-434e-b57d-284e759778cf.png)

### Expected behavior

The hub module should successfully run and register the cluster to hub project. When I try manually run the script it works by providing bash <scriptfilename.sh>

### Terraform Configuration

```hcl
locals {
  cluster_name = "xxx-demo-cluster"
}

resource "random_string" "suffix" {
  length  = 4
  special = false
  upper   = false
}

provider "google-beta" {
  version     = "~> 3.87.0"
  credentials = file(var.credentials_path)
  region      = var.region
}

data "google_compute_subnetwork" "subnetwork" {
  name    = var.subnetwork
  project = var.project_id
  region  = var.region
}

data "google_client_config" "default" {}

provider "kubernetes" {
  host                   = "https://${module.gke.endpoint}"
  token                  = data.google_client_config.default.access_token
  cluster_ca_certificate = base64decode(module.gke.ca_certificate)
}

module "gke" {
  source                  = "../../modules/beta-private-cluster-update-variant"
  project_id              = var.project_id
  name                    = "${local.cluster_name}-cluster-${random_string.suffix.result}"
  regional                = true
  region                  = var.region
  zones                   = var.zones
  network                 = var.network
  subnetwork              = var.subnetwork
  ip_range_pods           = var.ip_range_pods
  ip_range_services       = var.ip_range_services
  create_service_account  = false
  service_account         = var.compute_engine_service_account
  enable_private_endpoint = true
  enable_private_nodes    = true
  master_ipv4_cidr_block  = "172.18.10.0/28"

  master_authorized_networks = [
    {
      cidr_block   = data.google_compute_subnetwork.subnetwork.ip_cidr_range
      display_name = "VPC"
    },
  ]

  node_pools = [
    {
      name            = "apps-node-pool-01"
      machine_type    = "e2-medium"
      min_count       = 1
      max_count       = 1
      disk_size_gb    = 30
      disk_type       = "pd-standard"
      auto_upgrade    = false
      auto_repair     = false
      service_account = var.compute_engine_service_account
    },
    {
      name            = "ingress-node-pool-02"
      machine_type    = "e2-medium"
      min_count       = 1
      max_count       = 1
      disk_size_gb    = 30
      disk_type       = "pd-standard"
      image_type      = "COS"
      auto_repair     = false
      auto_upgrade    = false
      service_account = var.compute_engine_service_account
    },
    {
      name            = "egress-node-pool-03"
      machine_type    = "e2-medium"
      min_count       = 1
      max_count       = 1
      disk_size_gb    = 30
      disk_type       = "pd-standard"
      image_type      = "COS"
      auto_repair     = false
      auto_upgrade    = false
      service_account = var.compute_engine_service_account
    },
  ]

  node_pools_oauth_scopes = {
    all                  = ["https://www.googleapis.com/auth/cloud-platform"]
    apps-node-pool-01    = ["https://www.googleapis.com/auth/cloud-platform"]
    ingress-node-pool-02 = ["https://www.googleapis.com/auth/cloud-platform"]
    egress-node-pool-03  = ["https://www.googleapis.com/auth/cloud-platform"]
  }

  node_pools_metadata = {
    all = {}
    apps-node-pool-01 = {
      shutdown-script = file("${path.module}/data/shutdown-script.sh")
    }
    ingress-node-pool-02 = {}
    egress-node-pool-03  = {}
  }

  node_pools_labels = {
    all = {
      all-pools-example = true
    }
    apps-node-pool-01 = {
      apps-node-pool-01 = true
    }
    ingress-node-pool-02 = {
      ingress-node-pool-02 = true
    }
    egress-node-pool-03 = {
      egress-node-pool-03 = true
    }
  }


  node_pools_tags = {
    all = [
      "all-node-example",
    ]
    apps-node-pool-01 = [
      "apps-node-pool-01",
    ]
    ingress-node-pool-02 = [
      "ingress-node-pool-02",
    ]
    egress-node-pool-03 = [
      "egress-node-pool-03",
    ]
  }
}
module "hub" {

  source           = "../../modules/hub"
  project_id       = var.project_id
  cluster_name     = module.gke.name
  location         = module.gke.location
  cluster_endpoint = module.gke.endpoint
  hub_project_id   = var.hub_project_id

}
```

### Terraform Version

```sh
1.1.4
```

**Comment by guid1984** [comment]
> ### TL;DR
> the hub submodule is not working, specifically the script gke_hub_registration.sh. Getting permission denied error when terraform trying to execute the sh script via "terraform-google-modules/gcloud/google"
> 
> ![image](https://user-images.githubusercontent.com/8578605/150882855-e66a9b21-baa1-434e-b57d-284e759778cf.png)
> 
> ### Expected behavior
> The hub module should successfully run and register the cluster to hub project. When I try manually run the script it works by providing bash <scriptfilename.sh>
> 
> ### Observed behavior
> _No response_
> 
> ### Terraform Configuration
> ```terraform
> locals {
>   cluster_name = "xxx-demo-cluster"
> }
> 
> resource "random_string" "suffix" {
>   length  = 4
>   special = false
>   upper   = false
> }
> 
> provider "google-beta" {
>   version     = "~> 3.87.0"
>   credentials = file(var.credentials_path)
>   region      = var.region
> }
> 
> data "google_compute_subnetwork" "subnetwork" {
>   name    = var.subnetwork
>   project = var.project_id
>   region  = var.region
> }
> 
> data "google_client_config" "default" {}
> 
> provider "kubernetes" {
>   host                   = "https://${module.gke.endpoint}"
>   token                  = data.google_client_config.default.access_token
>   cluster_ca_certificate = base64decode(module.gke.ca_certificate)
> }
> 
> module "gke" {
>   source                  = "../../modules/beta-private-cluster-update-variant"
>   project_id              = var.project_id
>   name                    = "${local.cluster_name}-cluster-${random_string.suffix.result}"
>   regional                = true
>   region                  = var.region
>   zones                   = var.zones
>   network                 = var.network
>   subnetwork              = var.subnetwork
>   ip_range_pods           = var.ip_range_pods
>   ip_range_services       = var.ip_range_services
>   create_service_account  = false
>   service_account         = var.compute_engine_service_account
>   enable_private_endpoint = true
>   enable_private_nodes    = true
>   master_ipv4_cidr_block  = "172.18.10.0/28"
> 
>   master_authorized_networks = [
>     {
>       cidr_block   = data.google_compute_subnetwork.subnetwork.ip_cidr_range
>       display_name = "VPC"
>     },
>   ]
> 
>   node_pools = [
>     {
>       name            = "apps-node-pool-01"
>       machine_type    = "e2-medium"
>       min_count       = 1
>       max_count       = 1
>       disk_size_gb    = 30
>       disk_type       = "pd-standard"
>       auto_upgrade    = false
>       auto_repair     = false
>       service_account = var.compute_engine_service_account
>     },
>     {
>       name            = "ingress-node-pool-02"
>       machine_type    = "e2-medium"
>       min_count       = 1
>       max_count       = 1
>       disk_size_gb    = 30
>       disk_type       = "pd-standard"
>       image_type      = "COS"
>       auto_repair     = false
>       auto_upgrade    = false
>       service_account = var.compute_engine_service_account
>     },
>     {
>       name            = "egress-node-pool-03"
>       machine_type    = "e2-medium"
>       min_count       = 1
>       max_count       = 1
>       disk_size_gb    = 30
>       disk_type       = "pd-standard"
>       image_type      = "COS"
>       auto_repair     = false
>       auto_upgrade    = false
>       service_account = var.compute_engine_service_account
>     },
>   ]
> 
>   node_pools_oauth_scopes = {
>     all                  = ["https://www.googleapis.com/auth/cloud-platform"]
>     apps-node-pool-01    = ["https://www.googleapis.com/auth/cloud-platform"]
>     ingress-node-pool-02 = ["https://www.googleapis.com/auth/cloud-platform"]
>     egress-node-pool-03  = ["https://www.googleapis.com/auth/cloud-platform"]
>   }
> 
>   node_pools_metadata = {
>     all = {}
>     apps-node-pool-01 = {
>       shutdown-script = file("${path.module}/data/shutdown-script.sh")
>     }
>     ingress-node-pool-02 = {}
>     egress-node-pool-03  = {}
>   }
> 
>   node_pools_labels = {
>     all = {
>       all-pools-example = true
>     }
>     apps-node-pool-01 = {
>       apps-node-pool-01 = true
>     }
>     ingress-node-pool-02 = {
>       ingress-node-pool-02 = true
>     }
>     egress-node-pool-03 = {
>       egress-node-pool-03 = true
>     }
>   }
> 
> 
>   node_pools_tags = {
>     all = [
>       "all-node-example",
>     ]
>     apps-node-pool-01 = [
>       "apps-node-pool-01",
>     ]
>     ingress-node-pool-02 = [
>       "ingress-node-pool-02",
>     ]
>     egress-node-pool-03 = [
>       "egress-node-pool-03",
>     ]
>   }
> }
> module "hub" {
> 
>   source           = "../../modules/hub"
>   project_id       = var.project_id
>   cluster_name     = module.gke.name
>   location         = module.gke.location
>   cluster_endpoint = module.gke.endpoint
>   hub_project_id   = var.hub_project_id
> 
> }
> ```
> 
> ### Terraform Version
> ```shell
> 1.1.4
> ```
> 
> ### Additional information
> _No response_

using below works fine without any errors 

resource "null_resource" "gke_hub_registration_custom" {

 provisioner "local-exec" {

    command = "/bin/bash ${local.create_cmd_gke_entrypoint} ${local.create_cmd_gke_body}"
  }
  depends_on = [
    var.cluster_endpoint
  ] 
}

**Comment by guid1984** [comment]
> ### TL;DR
> the hub submodule is not working, specifically the script gke_hub_registration.sh. Getting permission denied error when terraform trying to execute the sh script via "terraform-google-modules/gcloud/google"
> 
> ![image](https://user-images.githubusercontent.com/8578605/150882855-e66a9b21-baa1-434e-b57d-284e759778cf.png)
> 
> ### Expected behavior
> The hub module should successfully run and register the cluster to hub project. When I try manually run the script it works by providing bash <scriptfilename.sh>
> 
> ### Observed behavior
> _No response_
> 
> ### Terraform Configuration
> ```terraform
> locals {
>   cluster_name = "xxx-demo-cluster"
> }
> 
> resource "random_string" "suffix" {
>   length  = 4
>   special = false
>   upper   = false
> }
> 
> provider "google-beta" {
>   version     = "~> 3.87.0"
>   credentials = file(var.credentials_path)
>   region      = var.region
> }
> 
> data "google_compute_subnetwork" "subnetwork" {
>   name    = var.subnetwork
>   project = var.project_id
>   region  = var.region
> }
> 
> data "google_client_config" "default" {}
> 
> provider "kubernetes" {
>   host                   = "https://${module.gke.endpoint}"
>   token                  = data.google_client_config.default.access_token
>   cluster_ca_certificate = base64decode(module.gke.ca_certificate)
> }
> 
> module "gke" {
>   source                  = "../../modules/beta-private-cluster-update-variant"
>   project_id              = var.project_id
>   name                    = "${local.cluster_name}-cluster-${random_string.suffix.result}"
>   regional                = true
>   region                  = var.region
>   zones                   = var.zones
>   network                 = var.network
>   subnetwork              = var.subnetwork
>   ip_range_pods           = var.ip_range_pods
>   ip_range_services       = var.ip_range_services
>   create_service_account  = false
>   service_account         = var.compute_engine_service_account
>   enable_private_endpoint = true
>   enable_private_nodes    = true
>   master_ipv4_cidr_block  = "172.18.10.0/28"
> 
>   master_authorized_networks = [
>     {
>       cidr_block   = data.google_compute_subnetwork.subnetwork.ip_cidr_range
>       display_name = "VPC"
>     },
>   ]
> 
>   node_pools = [
>     {
>       name            = "apps-node-pool-01"
>       machine_type    = "e2-medium"
>       min_count       = 1
>       max_count       = 1
>       disk_size_gb    = 30
>       disk_type       = "pd-standard"
>       auto_upgrade    = false
>       auto_repair     = false
>       service_account = var.compute_engine_service_account
>     },
>     {
>       name            = "ingress-node-pool-02"
>       machine_type    = "e2-medium"
>       min_count       = 1
>       max_count       = 1
>       disk_size_gb    = 30
>       disk_type       = "pd-standard"
>       image_type      = "COS"
>       auto_repair     = false
>       auto_upgrade    = false
>       service_account = var.compute_engine_service_account
>     },
>     {
>       name            = "egress-node-pool-03"
>       machine_type    = "e2-medium"
>       min_count       = 1
>       max_count       = 1
>       disk_size_gb    = 30
>       disk_type       = "pd-standard"
>       image_type      = "COS"
>       auto_repair     = false
>       auto_upgrade    = false
>       service_account = var.compute_engine_service_account
>     },
>   ]
> 
>   node_pools_oauth_scopes = {
>     all                  = ["https://www.googleapis.com/auth/cloud-platform"]
>     apps-node-pool-01    = ["https://www.googleapis.com/auth/cloud-platform"]
>     ingress-node-pool-02 = ["https://www.googleapis.com/auth/cloud-platform"]
>     egress-node-pool-03  = ["https://www.googleapis.com/auth/cloud-platform"]
>   }
> 
>   node_pools_metadata = {
>     all = {}
>     apps-node-pool-01 = {
>       shutdown-script = file("${path.module}/data/shutdown-script.sh")
>     }
>     ingress-node-pool-02 = {}
>     egress-node-pool-03  = {}
>   }
> 
>   node_pools_labels = {
>     all = {
>       all-pools-example = true
>     }
>     apps-node-pool-01 = {
>       apps-node-pool-01 = true
>     }
>     ingress-node-pool-02 = {
>       ingress-node-pool-02 = true
>     }
>     egress-node-pool-03 = {
>       egress-node-pool-03 = true
>     }
>   }
> 
> 
>   node_pools_tags = {
>     all = [
>       "all-node-example",
>     ]
>     apps-node-pool-01 = [
>       "apps-node-pool-01",
>     ]
>     ingress-node-pool-02 = [
>       "ingress-node-pool-02",
>     ]
>     egress-node-pool-03 = [
>       "egress-node-pool-03",
>     ]
>   }
> }
> module "hub" {
> 
>   source           = "../../modules/hub"
>   project_id       = var.project_id
>   cluster_name     = module.gke.name
>   location         = module.gke.location
>   cluster_endpoint = module.gke.endpoint
>   hub_project_id   = var.hub_project_id
> 
> }
> ```
> 
> ### Terraform Version
> ```shell
> 1.1.4
> ```
> 
> ### Additional information
> _No response_

It also works by modifying it like below 
module "gke_hub_registration" {
  source  = "terraform-google-modules/gcloud/google"
  version = "~> 2.1.0"

  platform                          = "linux"
  gcloud_sdk_version                = var.gcloud_sdk_version
  upgrade                           = true
  use_tf_google_credentials_env_var = var.use_tf_google_credentials_env_var
  module_depends_on                 = concat([var.cluster_endpoint], var.module_depends_on)

  create_cmd_entrypoint  = **"/bin/bash ${local.create_cmd_gke_entrypoint}"**
  create_cmd_body        = local.create_cmd_gke_body
  destroy_cmd_entrypoint = **"/bin/bash ${local.destroy_gke_entrypoint}"**
  destroy_cmd_body       = local.destroy_gke_body
}
